GitHubData/GitHubDataService.cs

```diff
@@ -155,30 +328,194 @@ public async Task<bool> ReleaseExists(string repoOwner, string repoName, string
         }
 
         var requestUri = $"repos/{repoOwner}/{repoName}/releases";
-        var response = await this.client.GetAsync<ReleaseModel[]>(requestUri);
+        var releases = await this.client.GetAsync<ReleaseModel[]>(requestUri);
 
-        if (response.statusCode != HttpStatusCode.OK)
+        if (releases is null)
         {
+            onError?.Invoke(new HttpRequestException($"The response from '{this.client.BaseUrl}/{requestUri}' returned null."));
             return false;
         }
 
-        var releaseExists = response.data?.Any(release =>
+        var releaseExists = releases.Any(release =>
         {
-            if (checkPreReleases is true)
+            if (checkPreReleases)
             {
                 return string.Equals(release.Name, releaseName, StringComparison.CurrentCultureIgnoreCase) &&
                        release.PreRelease;
             }
 
-            return string.Equals(release.Name, releaseName, StringComparison.CurrentCultureIgnoreCase);
-        }) ?? false;
+            return string.Equals(release.Name, releaseName, StringComparison.CurrentCultureIgnoreCase) &&
+                   release.PreRelease is false;
+        });
 
         // Cache the request result
         this.releaseNamesRequested.Add(releaseNameKey, releaseExists);
 
         return releaseExists;
     }
 
+    /// <inheritdoc/>
+    /// <returns>An asynchronous <b>string</b> data result of the file.</returns>
+    /// <exception cref="InvalidOperationException">
+    /// Thrown if any of the properties are empty or null.
```

Comment:
Refactor code docs to value below:

```suggestion
    /// Thrown if the following arguments are empty or null.
```